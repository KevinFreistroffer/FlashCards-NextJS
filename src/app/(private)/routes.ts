import { IRoute } from "@/_lib/definitions";

export const protectedRoutes: IRoute[] = [
  // {
  //   route: "/dashboard",
  //   text: "Dashboard",
  // },
];
