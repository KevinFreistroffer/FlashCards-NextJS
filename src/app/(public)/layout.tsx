import type { Metadata } from "next";
import { Inter } from "next/font/google";
import "../globals.css";
import Body from "../components/Body/Body";

const inter = Inter({ subsets: ["latin"] });

export const metadata: Metadata = {
  title: "Next-Drizzle-Auth",
  description: "",
};

export default function RootLayout({
  children,
}: Readonly<{
  children: React.ReactNode;
}>) {
  return (
    <html lang="en">
      <head>
        <script
          src="https://kit.fontawesome.com/4f00a527b1.js"
          crossOrigin="anonymous"
          async
        ></script>
      </head>
      <Body>{children}</Body>
    </html>
  );
}
