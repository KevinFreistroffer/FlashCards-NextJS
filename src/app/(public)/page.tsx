import Image from "next/image";
import Link from "next/link";
import "./styles.css";
import FlashCards from "../components/FlashCards/FlashCards";

export default function Home() {
  return (
    <main
      id={"main-screen"}
      className="flex min-h-screen justify-center mt-30 mb-20  p-24"
    >
      <FlashCards />
    </main>
  );
}
