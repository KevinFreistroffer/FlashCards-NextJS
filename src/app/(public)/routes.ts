import { IRoute } from "@/lib/definitions";
import { text } from "stream/consumers";

export const publicRoutes: IRoute[] = [
  {
    route: "/",
    text: "Home",
  },
  // {
  //   route: "/about",
  //   text: "About",
  // },
  // {
  //   route: "/signup",
  //   text: "Sign Up",
  // },
  // {
  //   route: "/login",
  //   text: "Login",
  // },
  // {
  //   route: "/form",
  //   text: "Form",
  // },
];
