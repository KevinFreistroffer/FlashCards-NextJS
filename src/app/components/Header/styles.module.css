.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1rem 2rem;
  background-color: #f8f9fa;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
}

.nav {
}

#menu-toggle {
  display: block;
  margin: auto;
  cursor: pointer;
}

.nav__ul {
  display: none;
}

@media only screen and (min-width: 769px) {
  #menu-toggle {
    display: none;
  }
  .nav__ul {
    display: flex;
  }
}
