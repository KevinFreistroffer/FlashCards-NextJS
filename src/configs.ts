export const configs = {
  title: "Flash Cards",
  description: "",
};

export default configs;
