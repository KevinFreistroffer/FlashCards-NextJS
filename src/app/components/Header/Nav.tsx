"use client";

import Link from "next/link";
import { usePathname, useRouter } from "next/navigation";
import React from "react";
import { publicRoutes } from "@/app/(public)/routes";
import styles from "./styles.module.css";

const Nav = ({
  toggleMenu,
}: {
  toggleMenu: React.Dispatch<React.SetStateAction<boolean>>;
}): JSX.Element => {
  const pathname = usePathname();
  return (
    <nav className={`${styles["nav"]}`}>
      <i
        id={styles["menu-toggle"]}
        onClick={() => toggleMenu((prev) => !prev)}
        className={"fa fa-bars"}
      ></i>
      <ul className={`${styles["nav__ul"]}`}>
        {publicRoutes.map(({ route, text }) => (
          <li key={route} className={`${styles["ul__li"]}  mr-3`}>
            <Link
              href={route}
              className={`${styles["ul__li"]} ${
                pathname === route ? "font-bold" : ""
              }  mr-3`}
            >
              {text}
            </Link>
          </li>
        ))}
      </ul>
    </nav>
  );
};

export default Nav;
