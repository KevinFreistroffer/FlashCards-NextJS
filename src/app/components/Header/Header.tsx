import React from "react";
import styles from "./styles.module.css";
import Link from "@/app/components/Header/Nav";
import { publicRoutes } from "@/app/(public)/routes";
import CustomLink from "@/app/components/Header/Nav";
import Nav from "@/app/components/Header/Nav";

interface IProps {
  title: string;
  menuVisible: boolean;
  toggleMenu: React.Dispatch<React.SetStateAction<boolean>>;
}

const Header = ({ menuVisible, toggleMenu, title }: IProps) => {
  return (
    <header
      className={`${styles["header"]} ${menuVisible ? "menu-visible" : ""}`}
    >
      <h1>{title}</h1>
      <Nav toggleMenu={toggleMenu} />
    </header>
  );
};

export default Header;
