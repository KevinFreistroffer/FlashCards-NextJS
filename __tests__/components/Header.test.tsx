import Header from "@/app/components/Header/Header";
import "@testing-library/jest-dom";
import { render, screen, within } from "@testing-library/react";
import configs from "../../src/configs";
import { publicRoutes } from "@/app/(public)/routes";
describe("Header", () => {
  const mockToggleMenu = jest.fn();
  it("renders a heading & menu", () => {
    render(
      <Header
        menuVisible={false}
        toggleMenu={mockToggleMenu}
        title={configs.title}
      />
    );
    screen.debug();

    const heading = screen.getByRole("heading", { level: 1 });
    const menu = screen.getByRole("navigation");
    const links = within(menu).getAllByRole("link");
    expect(heading).toHaveTextContent(configs.title);

    links.forEach((link, index) => {
      expect(link).toHaveTextContent(publicRoutes[index].text);
    });
  });
});
