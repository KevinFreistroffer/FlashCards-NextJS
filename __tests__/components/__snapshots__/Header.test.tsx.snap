// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot 1`] = `
<div>
  <header
    class="header "
  >
    <h1>
      Flash Cards
    </h1>
    <nav
      class="nav"
    >
      <i
        class="fa fa-bars"
        id="menu-toggle"
      />
      <ul
        class="nav__ul"
      >
        <li
          class="ul__li  mr-3"
        >
          <a
            class="ul__li   mr-3"
            href="/"
          >
            Home
          </a>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
